pynicotine/config.py

```diff
@@ -699,6 +699,30 @@ def _set_config(self):
                     self.sections[section][option] = use_speed_limit
                     continue
 
+                # Map legacy folder/user grouping modes (3.1.0)
+                if (section, option) in [
+                    ("searches", "group_searches"),
+                    ("transfers", "groupdownloads"),
+                    ("transfers", "groupuploads")
+                ]:
+                    mode = self.sections[section].get(option, "folder_grouping")
+
+                    if mode == "0":
+                        mode = "ungrouped"
+
+                    elif mode == "1":
+                        mode = "folder_grouping"
+
+                    elif mode == "2":
+                        mode = "user_grouping"
+
+                    # Verify mode validity
+                    elif mode not in {"ungrouped", "folder_grouping", "user_grouping"}:
+                        mode = "folder_grouping"
+
+                    self.sections[section][option] = mode
+                    continue
```

Comment:
This code block isn't currently reached. It should be moved down after line 728 (change `if (section, option) in [...]` to `for section, option in (...)`).

Reply:
Oh dear, the method has become too complex. Perhaps we need a dedicated `_migrate_config()` function, or such?

Reply:
Most likely, but don't include it in this PR.

Reply:
Looks like the linter agrees too

Reply:
Done in 778e2748d8ace661c2549eaf973a985f1d65162b